### PR TITLE
feat(QAB-164): Update s3 steps to use aws golang sdk v2

### PR DESCRIPTION
**Pull request by jordipuigbou**
*Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request.*

Please provide enough information so that others can review your pull request:

Explain the **details** for making this change. What existing problem does the pull request solve?

**Documentation**

Update Readme docs if needed.

**Test plan (required)**

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

**Code formatting**

**Closing issues**

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+.PHONY: download-tools
+download-tools:	## Download all required tools to validate and generate documentation, code analysis...
+	@echo "Installing tools on $(GO_PATH)/bin"
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.46.0
+	@echo "Tools installed"
+
+.PHONY: lint
+lint:	## Run static linting of source files. See .golangci.yml for options
+	golangci-lint $(LINTER_ARGS)
```
Comment (mariolg) [inline Makefile:27]: Add extra line.

File: steps/aws/s3/session_test.go [hunk 2]
```diff
@@ -67,11 +68,6 @@ func TestNewS3Session(t *testing.T) {
 		newS3SessionErr error
 		wantErr         bool
 	}{
-		{
-			name:            "New session error",
-			newS3SessionErr: fmt.Errorf("new s3 session error"),
-			wantErr:         true,
-		},
```
Comment (mariolg) [inline steps/aws/s3/session_test.go:74]: This tests was not testable?

**Review by mariolg — COMMENTED** [review]
Have a look at my comments/questions?

**Review by ricardogarfe — APPROVED** [review]

File: steps/aws/s3/session_test.go (hunk 2 above)
Comment (jordipuigbou) [reply steps/aws/s3/session_test.go]: Session configuration changes a bit with the new sdk version and client creation does not returns error. The only way to get error in this function is when LoadDefaultConfig fails but not easy to achieve without mock. 
I think that in this case we could skip this error check.

File: .github/workflows/sonarcloud.yml [hunk 3]
```diff
@@ -30,12 +30,12 @@ jobs:
       - name: Golangci increment
         if: "!contains(github.ref, 'master')"
         run: |
-          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.44.2
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.46.0
```
[the remark this answers is not in the document]
Comment (jordipuigbou) [reply .github/workflows/sonarcloud.yml]: Done

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	go test ./test/acceptance -v --godog.format=$(GODOG_FORMAT)
+
+.PHONY: test-run-tag
+test-run-tag:	## Run feature from tag using variable TAG='<@tag_name>'
+	go test ./test/acceptance -v --godog.tags=$(TAG) --godog.format=$(GODOG_FORMAT)
+
+.PHONY: download-tools
+download-tools:	## Download all required tools to validate and generate documentation, code analysis...
+	@echo "Installing tools on $(GO_PATH)/bin"
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.46.0
```
[the remark this answers is not in the document]
Comment (jordipuigbou) [reply Makefile]: Done

**Review by mariolg — APPROVED** [review]
LGTM

**Pull request merged**